### ISSUE TITLE
Reproducing the paper results

### ISSUE THREAD

**Issue by nctamer** [Yujia-Yan/Skipping-The-Frame-Level#26]
Hi, thanks for the great work. I am trying to reproduce the results published in paper following the [training instructions](https://github.com/Yujia-Yan/Transkun?tab=readme-ov-file#training), but after 45 epochs, training remains very fuzzy and F1 score on the validation set is ~0.36 and not improving. I tried multiple runs with and without augmentation. (Using the configuration generated by moduleconf as recommended, on a single L40S with batch size of 16). 

I thought the training script might not be up-to-date, but I would also appreciate to learn about the specific arguments if anyone else was able to reproduce the training with the script provided.

**Comment by Yujia-Yan** [comment]
Hi, could you show me the training curves? Have you performed sampling rate conversion for 2017/2018 in the dataset?

**Comment by nctamer** [comment]
Hi, thanks for the fast reply! Yes, I did the sampling rate conversion on the Maestro folder using torch audio resampler (`torchaudio.transforms.Resample(orig_freq=fs, new_freq=44100, resampling_method='sinc_interp_hann', lowpass_filter_width=64)`). You can find a report for a problematic part of the training in the attached pdf (between epochs 20-34, where the val metrics stopped improving. The performance remained around this range afterwards.). Thank you!

[Epochs20-34 | transkun – Weights & Biases.pdf](https://github.com/user-attachments/files/18528894/Epochs20-34.transkun.Weights.Biases.pdf)

**Comment by Yujia-Yan** [comment]
The training recall curve frequently touches very low values. It exactly fits the pattern where 2017/2018 are in a different sampling rate thus midi and audio segments are totally irrelevant . Are converted .wav files stored at the same path as the original ones?

**Comment by nctamer** [comment]
Hi YuJia, that was such an amazing support! The .wav files were stored in the same folder, but are right that the problem was with those resampled files! It was due to mismatch in gain, the files after resampling were in float (-1,1 range) whereas the readAudioSlice expects differently). My kudos for identifying that from this report only! Issue closed in less than an hour!:)
